ESPLoader.js

```diff
@@ -1586,6 +1587,26 @@ class ESPLoader {
         this.log("Device: "+((flashid >> 8) & 0xff).toString(16) + flid_lowbyte.toString(16));
         this.log("Detected flash size: " + this.DETECTED_FLASH_SIZES[flid_lowbyte]);
     }
+
+    hard_reset = async() => {
+        this.transport.setRTS(true);  // EN->LOW
+        await this._sleep(100);
+        this.transport.setRTS(false);
+    }
+
+    soft_reset = async() => {
+        if (!this.IS_STUB) {
+            // 'run user code' is as close to a soft reset as we can do
+            this.flash_begin(0, 0);
+            this.flash_finish(false);
+        } else if (this.chip.CHIP_NAME != "ESP8266") {
+            throw("Soft resetting is currently only supported on ESP8266");
```

Comment:
Error reporting #26 has been merged. Shall we have this raise?
```suggestion
            throw new ESPError("Soft resetting is currently only supported on ESP8266");
```